parcel-bundler/parcel
# Improve elm compiler error output

# ↪️ Pull Request

Fixes https://github.com/parcel-bundler/parcel/issues/7957

This PR adds changes to the `transformer-elm` that makes compiler output look more like they used to in Parcel v1.
The way it is currently is hard to read in many terminals with a dark background.

## 💻 Examples

This implements to following change, when any compilation error happens for Elm code. The output is more readable because it is logged as in the `message` of the diagnostic instead of the `stack` part.

### Before:
![image](https://user-images.githubusercontent.com/5880236/164766537-5f50556a-849a-4890-a097-65ae76e11d70.png)

### After:
![image](https://user-images.githubusercontent.com/5880236/164766668-87e1b604-91ed-41d2-a096-b0437c01eba8.png)


## 🚨 Test instructions

This PR also adds an entry in the `packages/examples` for folder elm for testing. It was useful for me and I thought that was the way to go from reading the `Contributing.md` file. If that example is not useful to anyone but me I can remove it from this PR if anyone lets me know.

Testable this way:
- cd into `packages/examples/elm`
- run `yarn install`
- delete random lines from `src/Main.elm` until the error output appears in the console.
- run `yarn demo`

## ✔️ Bonus

While this change improves the error output in the terminal, it kind of messes up the one in the browser.

With this change:

![image](https://user-images.githubusercontent.com/5880236/164769262-1b1b7992-6640-4523-815d-7a1f968b9915.png)

Before this change (correctly formatted but hard to read if background is dark also a gray tone):

![img](https://user-images.githubusercontent.com/23027708/142491398-59f73846-0df7-43e9-86c0-1fe679439232.png)

This is due to the browser display not respecting white space for formatting.

Would it be possible to add `white-space: pre;` to the `.error-message` class? It seems that whitespace is preserved when logging things as `stack` (but hard to read on dark background) but not for the `message` part of a throwable diagnostic. I think it would make sense to add this because otherwise you get things that look differently in the terminal and in the browser.
